markdown/3.x/en/ff-communication.md

```diff
@@ -27,4 +27,19 @@ correctly.
 
 **NOTE**: Although `version` defaults to `7.2` when not set, it is recommended to set it as default value may change with feature releases.
 
-For more information, see the [API reference](/api/3.x/ff-communication#tab=api).
\ No newline at end of file
+For more information, see the [API reference](/api/3.x/ff-communication#tab=api).
+
+### Single hit redirect deprecation
+Due to the feature's limitations described below, it will be removed in a future version.
+
+In order to prevent infinite redirection after getting back from a product page to the search results page, an internal `ff-no-redirect` flag is toggled after the first redirection takes place.
+With this flag toggled Web Components will not redirect the next search, even if there is only one record returned back from FACT-Finder.
+However, the next search will not necessarily be the one that was issued after returning to the search results page.
+If the user provides an `ff-searchbox` component on the product page, the next search could also be issued from there.
+Single hit redirect in this case will not work because of the `ff-no-redirect` flag is toggled on and as a result the user will land on the search result page.
+After that, the `ff-no-redirect` flag is turned off, and the next search request will be redirected, as expected.
+
+The other reason is that e-commerce platforms using Web Components are, in the most situations, not SPA (single page application). That force developers to manage routing
+on application frontend to provide redirection, for instance from home page to search result page. For more details about routing, please refer to [Routing in non-SPA websites](documentation/3.x/routing).
```

Comment:
I totally missed this section....
Consider using one line per sentence. This makes it easier to target annotations.
```suggestion
The other reason is that e-commerce platforms using Web Components are, in most cases, not an SPA (single page application).
This kind of application requires developers to manage routing in the application's frontend to provide redirection, for instance from the home page to a search result page.
For more details about routing, please refer to [Routing in non-SPA websites](documentation/3.x/routing).
```